lib/rspec/matchers/built_in/yield.rb

```diff
@@ -326,7 +337,7 @@ def negative_failure_reason
           end
```

Comment:
Here in `negative_failure_reason` I am realizing that `actual_formatted` is evaluated, which means that it will include the yielded args as they exist at the end.  For parity with `positive_failure_reason` it would be good for it to show the args as they were when yielded.

I think the simplest solution might be to evaluate `actual_formatted` at yield time and store in a variable, which can be used for both the positive and negative failure messages.

Also: I believe that there is not test coverage for this, to enforce that the failure messages for `yield_with_args` include the formatted versions of what was yielded at the time they were yielded.  Can you add test coverage?